### ISSUE TITLE
Lat-by-lat RCE for time-invariant insolation at specific time of year

### ISSUE THREAD

**Issue by spencerahill** [brian-rose/climlab#89]
This is a bit of a funny use case, so I can understand if it's not easily done out-of-the-box, BUT: I would like to compute RCE solutions as a function of latitude given insolation at different points in the annual cycle, but with the insolation itself time-invariant.

For example, set the insolation to northern hemisphere solstice, and then run the RCE models to steady-state, i.e. a year or so.

What would be the simplest way of doing this?  If I use `DailyInsolation`, then it evolves in time as the model is integrated forward in time.  Is there an option to freeze it at a given point in the year?

A brute-force workaround would seem to be to get the S0, zenith angle, and any other needed parameters from the `daily_insolation` function, and then perform separate column RCE integrations for each of N latitude bands with `FixedInsolation`.  If there's not a simpler way, would this work?

(Motivation: for the Hadley cells, it is useful to know what the zonal wind field would be if there were no large-scale circulation, which is the zonal wind field in gradient balance with the RCE temperature field.  Traditionally this has been done for dry atmospheres using Newtonian cooling toward a prescribed temperature profile, but I want to compute it explicitly throughout the seasonal cycle including moisture and more realistic radiation.  So I need to compute the steady-state RCE solutions at different latitudes and at different times of the year).

**Comment by brian-rose** [comment]
I created climlab specifically to make the funny use cases easier to do!

But this one is actually easy. If you're holding the insolation fixed in time, then you don't need a separate Process to represent insolation. You just need to set the appropriate inputs to the radiation process and leave it that way.

I wrote this example, somewhat more elegant than your proposed brute force method:
```
num_lat = 60; num_lev = 50   # grid size
day_of_year = 80.  # days since Jan 1
import climlab
state = climlab.column_state(num_lev=num_lev, num_lat=num_lat)
conv = climlab.convection.ConvectiveAdjustment(name='Convection', state=state, adj_lapse_rate=6.5)
insolation = climlab.solar.insolation.daily_insolation(lat=conv.lat, day=day_of_year)
rad = climlab.radiation.RRTMG(name='Radiation', state=state, insolation=insolation.values)
rcm = climlab.couple([rad, conv], name='Radiative-Convective Model with fixed insolation')
rcm.step_forward()  # single timestep
```

This fails due to an internal broadcasting error affecting the `coszen` array that is passed to the RRTMG fortran code. This needs to be fixed; I'll open a new issue for that.

However when I tested this I found out that **the above code fails due to a previously unknown bug**.

**Comment by brian-rose** [comment]
A workaround is, along the lines of your brute force suggestion, to get an array of insolation values like
```
insolation = climlab.solar.insolation.daily_insolation(lat=my_lat_array, day=day_of_year)
```
and loop through this array, creating a single-column model for each latitude point, and setting the `insolation` input parameter for the RRTMG object appropriately.

**Comment by brian-rose** [comment]
Once #90 is fixed, then the example code above should work as expected.

Please continue to bring up your funny use cases, because it really helps uncover these nasty bugs!

**Comment by spencerahill** [comment]
Thanks a lot for diving in to find this bug.  I'd be happy to contribute a quick fix...I'll try to get a PR in soon.  That sounds easier to me than the workaround!

**Comment by brian-rose** [comment]
This issue is still open after merging #93 because the example script at the top of the issue now fails with a segfault, presumably associated with #95

**Comment by brian-rose** [comment]
As noted in #95, the workaround is to use a smaller grid, e.g.:
```
num_lat = 30; num_lev = 50   # grid size
day_of_year = 80.  # days since Jan 1
import climlab
state = climlab.column_state(num_lev=num_lev, num_lat=num_lat)
conv = climlab.convection.ConvectiveAdjustment(name='Convection', state=state, adj_lapse_rate=6.5)
insolation = climlab.solar.insolation.daily_insolation(lat=conv.lat, day=day_of_year)
rad = climlab.radiation.RRTMG(name='Radiation', state=state, insolation=insolation.values)
rcm = climlab.couple([rad, conv], name='Radiative-Convective Model with fixed insolation')
rcm.step_forward()  # single timestep
```

**Comment by brian-rose** [comment]
This is now fixed with #98 

In other words, the thing that I said was easy at the top of this thread is now easy, after fixing several different bugs :-)

**Comment by brian-rose** [comment]
closing since the example script now works as expected.
